src/commands/context/tasks.ts

```diff
@@ -29,62 +30,142 @@ export class ContextCommandTasks {
+      await this.parent.getRemoteConfigManager().modify(async remoteConfig => {
+        const localConfig = this.parent.getLocalConfig();
+        const localDeployments = localConfig.deployments;
+        ctx.config.clusters = Object.keys(remoteConfig.clusters);
+        localDeployments[localConfig.currentDeploymentName].clusters = ctx.config.clusters;
+        localConfig.setDeployments(localDeployments);
 
-      let currentDeploymentName = argv[flags.namespace.name];
-      let clusters = Templates.parseClusterAliases(argv[flags.clusterName.name]);
-      let contextName = argv[flags.context.name];
+        const contexts = Templates.parseCommaSeparatedList(configManager.getFlag(flags.context));
 
-      const kubeContexts = await this.parent.getK8().getContexts();
+        for (let i = 0; i < ctx.config.clusters.length; i++) {
+          const cluster = ctx.config.clusters[i];
+          const context = contexts[i];
 
-      if (isQuiet) {
-        const currentCluster = await this.parent.getK8().getKubeConfig().getCurrentCluster();
-        if (!clusters.length) clusters = [currentCluster.name];
-        if (!contextName) contextName = await this.parent.getK8().getKubeConfig().getCurrentContext();
+          // If a context is provided use it to update the mapping
+          if (context) {
+            localConfig.clusterContextMapping[cluster] = context;
+          }
 
-        if (!currentDeploymentName) {
-          const selectedContext = kubeContexts.find(e => e.name === contextName);
-          currentDeploymentName = selectedContext && selectedContext.namespace ? selectedContext.namespace : 'default';
-        }
-      } else {
-        if (!clusters.length) {
-          const prompt = flags.clusterName.prompt;
-          const unparsedClusterAliases = await prompt(task, clusters);
-          clusters = Templates.parseClusterAliases(unparsedClusterAliases);
+          // In quiet mode use the currently selected context to update the mapping
+          else if (isQuiet) {
+            //default
+            localConfig.clusterContextMapping[cluster] = this.parent.getK8().getKubeConfig().getCurrentContext();
+          }
+
+          // Prompt the user to select a context if mapping value is missing
+          else if (!localConfig.clusterContextMapping[cluster]) {
+            const kubeContexts = this.parent.getK8().getContexts();
+            localConfig.clusterContextMapping[cluster] = await flags.context.prompt(
+              task,
+              kubeContexts.map(c => c.name),
+              cluster,
+            );
+          }
         }
-        if (!contextName) {
-          const prompt = flags.context.prompt;
-          contextName = await prompt(
-            task,
-            kubeContexts.map(c => c.name),
-          );
+        this.parent.logger.info('Update local configuration...');
+        await localConfig.write();
+      });
+    });
+  }
+
+  selectContext(argv) {
```

Comment:
would be good to make this method more modular, it is a bit long and has opportunities to be broken down